enclave_wrangler/utils.py

```diff
@@ -211,51 +217,109 @@ def handle_paginated_request(first_page_url: str, verbose=False, called_by: str=
     return results
 
 
+def get_objects_df(object_type, outdir: str = None, save_csv=True, save_json=True):
+    """Get objects as dataframe"""
+    # Get objects
+    objects: List[Dict] = make_objects_request(object_type, return_type='data', outdir=outdir)
+
+    # Parse
+    # Get rid of nested 'properties' key, and add 'rid' in with the other fields
+    df = pd.DataFrame()
+    if objects:
+        results = [{**x['properties'], **{'rid': x['rid']}} for x in objects]
+        df = pd.DataFrame(results).fillna('')
+
+    # Save
+    if outdir:
+        outdir = outdir if outdir else os.path.join(OUTDIR_OBJECTS, object_type)
+        outpath = os.path.join(outdir, 'latest.csv')
+        if not os.path.exists(outdir) and (save_json or save_csv):
+            os.mkdir(outdir)
+        # - csv
+        if save_csv:
+            df.to_csv(outpath, index=False)
+        # - json
+        if save_json:
+            with open(outpath.replace('.csv', '.json'), 'w') as f:
+                json.dump(objects, f)
+    return df
+
+
+# todo's from b4 refactor combining make_objects_request() and get_objects_by_type() 2023/03/15
+#   1. Need to find the right object_type, then write a wrapper func around this to get concept sets
+#     - To Try: CodeSystemConceptSetVersionExpressionItem, OMOPConcept, OMOPConceptSet, OMOPConceptSetContainer,
+#     OmopConceptSetVersionItem
+#   2. connect to `manage` table and get since last datetime. for now, use below as example
+# noinspection PyUnboundLocalVariable
 def make_objects_request(
```

Comment:
@Sigfried I finished refactoring `get_objects_by_type()` into `make_objects_request()`. This mainly adds `since_datetime` and `query_params`.

Reply:
I didn't see any import errors, but I wasn't able to check if things still work because object API still broken: 
- #279 

Would have been great to run some unit tests after this refactor, but we're not at that point.

Reply:
the 404's have since been fixed, and we've also refactored since_datetime into query_params, and put it in its own function